ethereum/core/src/main/java/org/hyperledger/besu/ethereum/trie/bonsai/storage/flat/FlatDbStrategyProvider.java

```diff
@@ -50,10 +52,15 @@ public void loadFlatDbStrategy(final SegmentedKeyValueStorage composedWorldState
     // if  flatDbMode is not loaded or has changed, reload flatDbStrategy
     if (this.flatDbMode == null || !this.flatDbMode.equals(newFlatDbMode)) {
       this.flatDbMode = newFlatDbMode;
+      final boolean isCodeStoredByHashEnabled =
+          dataStorageConfiguration.getUnstable().getBonsaiCodeStoredByCodeHashEnabled();
+      LOG.info(
+          "Bonsai flat db mode with code stored using code hash enabled = {}",
+          isCodeStoredByHashEnabled);
```

Comment:
the risk of driving this from command line only is that we can fall out of consensus if the user changes this config parameter without resyncing.  It isn't fun to plumb in the check  but we should try to derive the codehash vs accounthash as it exists in the database and only allow the cli parameter to affect an as-yet-unsynced node.

we already have the notion of using `loadFlatDbStrategy(..)` in BonsaiWorldStateKeyValueStorage, and it will override the CLI supplied flat db strategy based on what it finds.  We can work this check into that method and ensure that we use the correct code storage strategy based on what is in the db, and perhaps log a big ugly warning message if the CLI param does not match the db storage

Reply:
Added detection code in the `FlatDbStrategyProvider` which will warn if there is an inconsistency and use the strategy in the database.